aiohttp/http_writer.py

```diff
@@ -118,14 +121,58 @@ def _writelines(
         else:
             transport.writelines(chunks)  # type: ignore[arg-type]
 
+    def _write_chunked_payload(
+        self, chunk: Union[bytes, bytearray, "memoryview[int]", "memoryview[bytes]"]
+    ) -> None:
+        """Write a chunk with proper chunked encoding."""
+        chunk_len_pre = f"{len(chunk):x}\r\n".encode("ascii")
+        self._writelines((chunk_len_pre, chunk, b"\r\n"))
+
+    def _send_headers_with_payload(
+        self,
+        chunk: Union[bytes, bytearray, "memoryview[int]", "memoryview[bytes]"],
+        is_eof: bool,
+    ) -> None:
+        """Send buffered headers with payload, coalescing into single write."""
+        # Mark headers as written
+        self._headers_written = True
+        headers_buf = self._headers_buf
+        self._headers_buf = None
+
+        if TYPE_CHECKING:
```

Comment:
Is this check any more efficient than the assert? I'm also not sure if `-O` actually removes TYPE_CHECKING checks or not.

Reply:
orders of magnitude faster `{'if_TYPE_CHECKING_duration': -1.7500133253633976e-06, 'assert_TYPE_CHECKING_duration': 0.21601537498645484}`